src/util/misc/matrix.ts

```diff
@@ -277,14 +277,17 @@ export const calcDimensionsMatrix = ({
   skewX = 0 as TDegree,
   skewY = 0 as TDegree,
 }: TScaleMatrixArgs) => {
-  return multiplyTransformMatrixArray(
-    [
-      createScaleMatrix(flipX ? -scaleX : scaleX, flipY ? -scaleY : scaleY),
-      skewX && createSkewXMatrix(skewX),
-      skewY && createSkewYMatrix(skewY),
-    ],
-    true
+  let matrix = createScaleMatrix(
+    flipX ? -scaleX : scaleX,
+    flipY ? -scaleY : scaleY
   );
```

Comment:
> I m not sure is worth to create it rather than returning the iMatrix as a reference and then check by strict equality.

I thought about that but it's dangerous in fabric returning the same reference, as the `iMatrix` reference could be mutated with terrible consequences

Reply:
iMatrix is frozen and will throw if you try to mutate it, we can't return it

Reply:
but maybe it is quicker to create it than to clone it